### PR TITLE
WIP: Answer parsing feature, automatic paging

### PR DESCRIPTION
**Pull request by apurvis**
@jarthod - not ready to merge but curious if you had any feedback about this feature.  My use cases don't really use some of SG's more wacky ways of encoding data so definitely good to have any other input.

http://apihelp.surveygizmo.com/help/article/link/surveyresponse-per-question

### CODE REVIEW

**Comment by jarthod** [comment]
Interesting, that could make it easier to use indeed, but people won't think about using this unless we push it in the documentation (with output examples, etc). That would need a lot of testing though but it could become the default output format in a future version.

**Comment by apurvis** [comment]
pushed a better arrangement plus some decent specs to the PR (along with better documentation)

As a guard against breakage, right now it will exception out on any `answer` format it doesn't recognize.

File: README.md [hunk 1]
```diff
@@ -96,6 +94,19 @@ responses = SurveyGizmo::API::Response.all(
     }
   ]
 )
+# Retrieve all responses for a given survey.
+# Note that this may not be a good idea for surveys with very large numbers of responses!
+responses = SurveyGizmo::API::Response.all(all_pages: true, survey_id: 12345)
+# If you want the gem to handle paging for you, use the :all_pages option and process your pages in a block
```
Comment (apurvis) [inline README.md]: @jarthod here's another new feature - auto pagination

File: lib/survey_gizmo/rest_response.rb [hunk 2]
```diff
@@ -1,91 +1,100 @@
...
+module SurveyGizmo
+  class RestResponse
+    attr_accessor :raw_response
+    attr_accessor :parsed_response
 
-  def initialize(http_response)
-    @raw_response = http_response
-    @parsed_response = http_response.parsed_response
+    def initialize(http_response)
+      fail RateLimitExceededError if http_response.code == 429
```
Comment (apurvis) [inline lib/survey_gizmo/rest_response.rb]: this should make rate limit errors easier to handle

File: lib/survey_gizmo/rest_response.rb [hunk 3]
```diff
@@ -1,91 +1,100 @@
 # This class normalizes the response returned by Survey Gizmo, including validation.
 
-class RestResponse
-  attr_accessor :raw_response
-  attr_accessor :parsed_response
+module SurveyGizmo
```
Comment (apurvis) [inline lib/survey_gizmo/rest_response.rb]: for some reason this class wasn't even in the module - my bad.

**Pull request closed without merging**